### ISSUE TITLE
[Forwardport v2.7] [BUG] Rancher fails to install on 22.04 Ubuntu with error - `mkdir: cannot create directory '/sys/fs/cgroup/init': Read-only file system`

### ISSUE THREAD

**Issue by rancherbot** [rancher/rancher#39283]
This is a forwardport issue for https://github.com/rancher/rancher/issues/39281, automatically created via rancherbot by @sowmyav27

Original issue description:

**Rancher Server Setup**
- Rancher version: 2.6-head `f10a449b6` or 2.7-head 
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1 using - 1.3.15 or 1.4.0-rc3

**Describe the bug**
Rancher fails to install on 22.04 Ubuntu with error - `mkdir: cannot create directory '/sys/fs/cgroup/init': Read-only file system`

**To Reproduce**

- Deploy RKE1 -3 nodes all roles cluster using ubuntu 22.04 AMI
- Follow rancher docs to install cert manager and Rancher.
- Rancher is in a crashloop
- Error seen on rancher pods: `mkdir: cannot create directory '/sys/fs/cgroup/init': Read-only file system`

**Expected Result**
Rancher should come up successfully.

**Workaround**
- On the instance ran these commands
```
# echo 'GRUB_CMDLINE_LINUX=systemd.unified_cgroup_hierarchy=false' > /etc/default/grub.d/cgroup.cfg
# update-grub
```
- Rebooted the node, before creating a Private AMI from this to then use our automation scripts to install Rancher.
- And rancher 2.6-head provisioned successfully.

**Comment by adolfomaltez** [comment]
I got this error on Debian 11.5.
Try the alternative solution.
I confirm that it works.

**Comment by kinarashah** [comment]
Available to test with `v2.7-head` once https://drone-publish.rancher.io/rancher/rancher/8437 passes.

**Comment by sowmyav27** [comment]
On 2.7-head with the fix, was able to verify Rancher provisions on an RKE1 k8s local cluster using Ubuntu 22.04 AMI successfully.
